### PR TITLE
Build Android on Linux

### PR DESCRIPTION
**Pull request by mattjohnsonpint**
- Build Android targets on Linux
- Remove OS solution filters (.slnf files).  We don't need them now.
- Update build scripts and yaml

#skip-changelog

### CODE REVIEW

**Comment by bruno-garcia** [comment]
> * Remove OS solution filters (.slnf files).  We don't need them now.

Do we need them for iOS? And Windows-only stuff like WinForms and WPF sample? #1036

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -61,38 +54,29 @@ jobs:
...
-          run: dotnet workload install maui-android maui-windows
+          run: 'dotnet workload install maui-android maui-windows --temp-dir "${{ runner.temp }}"'
         - name: Install .NET Workloads (macOS)
           if: startsWith(matrix.os, 'macos')
-          run: dotnet workload install maui-android maui-ios maui-maccatalyst
+          run: 'dotnet workload install maui-android maui-ios maui-maccatalyst --temp-dir "${{ runner.temp }}"'
 
         - name: Build
-          run: dotnet build ${{ matrix.slnf }} -c Release /p:CopyLocalLockFileAssemblies=true
+          run: dotnet build -c Release /p:CopyLocalLockFileAssemblies=true
```
Comment (bruno-garcia) [inline .github/workflows/build.yml:71]: Could we set this property here: https://github.com/getsentry/sentry-dotnet/blob/main/Directory.Build.props#L20-L22

This way we don't need to remember to pass it in different places: https://github.com/getsentry/sentry-dotnet/search?q=p%3ACopyLocalLockFileAssemblies

Comment (mattjohnsonpint) [reply .github/workflows/build.yml:71]: No, because we don't want to do that in the CodeQL build.

**Comment by mattjohnsonpint** [comment]
> > * Remove OS solution filters (.slnf files).  We don't need them now.
> 
> Do we need them for iOS? And Windows-only stuff like WinForms and WPF sample? #1036

I'm investigating into another way to deal with that.  I don't like using solution filters for this.

**Review by SimonCropp — APPROVED** [review]

**Pull request merged**